core-bundle/src/DependencyInjection/Configuration.php

```diff
@@ -875,6 +875,10 @@ static function (array $allowedProperties): array {
                         )
                     ->end()
                 ->end()
+                ->integerNode('max_header_size')
+                    ->info('The CSP header can get pretty long when adding lots of automatically calculated hashes. Contao will automatically remove signatures in order to prevent the CSP header to exceed the configured limit. This will affect the functionality of affected URLs but ensure you do not run into 500 server errors.')
+                    ->defaultValue(3072) // Most browsers have a limit of 8-16kb for **all** headers in total
```

Comment:
This describes the technical background of the pull request rather than the `max_header_size` parameter, which is not very helpful for someone who wants to adjust the parameter and maybe needs some pointers, is it?

```suggestion
                    ->info('Do not increase this value beyond the header buffer size of your web server, as this will result in a 400 Bad Request server error.')
                    ->defaultValue(3072)
```

Reply:
Are you sure it would end in a 400? Sounds like it should be a 500 as it's not the client's fault?

Reply:
I haven‘t tried it, but it‘s what the Nginx docs say:

> A request line cannot exceed the size of one buffer, or the 414 (Request-URI Too Large) error is returned to the client. A request header field cannot exceed the size of one buffer as well, or the 400 (Bad Request) error is returned to the client.

https://nginx.org/en/docs/http/ngx_http_core_module.html#large_client_header_buffers

Reply:
Those are request headers. We are talking about response headers :) Pretty sure it would be something else in our case. Would you mind trying it out on one of your nginx setups? 😇

Reply:
I have updated my suggestion accordingly.